tests/src/test/spark350/scala/com/nvidia/spark/rapids/ToPrettyStringSuite.scala

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*** spark-rapids-shim-json-lines
+{"spark": "350"}
+spark-rapids-shim-json-lines ***/
+package com.nvidia.spark.rapids
+
+import ai.rapids.cudf.{ColumnVector, DType, HostColumnVector}
+import collection.JavaConverters._
+import com.nvidia.spark.rapids.Arm._
+import com.nvidia.spark.rapids.shims.GpuToPrettyString
+
+import org.apache.spark.sql.catalyst.InternalRow
+import org.apache.spark.sql.catalyst.expressions.{Literal, NamedExpression, ToPrettyString}
+import org.apache.spark.sql.catalyst.util.{ArrayBasedMapData, ArrayData}
+import org.apache.spark.sql.types.{ArrayType, DataType, DataTypes, MapType, StructField, StructType}
+import org.apache.spark.sql.vectorized.ColumnarBatch
+import org.apache.spark.unsafe.types.UTF8String
+
+class ToPrettyStringSuite extends GpuUnitTests {
```

Comment:
Could you add tests for other types, such as decimal, float/double, int types, string, covering null values. I noticed that we have different behavior for null decimals through manual testing, where Spark shows `NULL` and on GPU we show an empty string instead.

## CPU

```
scala> spark.sql("select ss_ext_sales_price from store_sales").show
+------------------+
|ss_ext_sales_price|
+------------------+
|            817.20|
|            117.04|
|           1189.20|
|            239.12|
|           1260.87|
|            185.13|
|            462.95|
|            122.94|
|             31.80|
|            111.29|
|              NULL|
|            131.25|
|            858.00|
|           1247.38|
|           3011.16|
|           2599.80|
|           2565.16|
|            343.06|
|           4918.32|
|             29.56|
+------------------+
only showing top 20 rows
```

## GPU 

```
scala> spark.sql("select ss_ext_sales_price from store_sales").show
23/09/21 16:46:04 WARN GpuOverrides: 
!Exec <CollectLimitExec> cannot run on GPU because the Exec CollectLimitExec has been disabled, and is disabled by default because Collect Limit replacement can be slower on the GPU, if huge number of rows in a batch it could help by limiting the number of rows transferred from GPU to CPU. Set spark.rapids.sql.exec.CollectLimitExec to true if you wish to enable it
  @Partitioning <SinglePartition$> could run on GPU
  *Exec <ProjectExec> will run on GPU
    *Expression <Alias> toprettystring(ss_ext_sales_price#15, Some(UTC)) AS toprettystring(ss_ext_sales_price)#251 will run on GPU
      *Expression <ToPrettyString> toprettystring(ss_ext_sales_price#15, Some(UTC)) will run on GPU
    *Exec <FileSourceScanExec> will run on GPU

+------------------+
|ss_ext_sales_price|
+------------------+
|            817.20|
|            117.04|
|           1189.20|
|            239.12|
|           1260.87|
|            185.13|
|            462.95|
|            122.94|
|             31.80|
|            111.29|
|                  |
|            131.25|
|            858.00|
|           1247.38|
|           3011.16|
|           2599.80|
|           2565.16|
|            343.06|
|           4918.32|
|             29.56|
+------------------+
only showing top 20 rows
```

Reply:
I also found some formatting differences with float and decimal that we should at least document if we cannot easily produce the same output, and file a follow-on issue.

```
scala> spark.conf.set("spark.rapids.sql.enabled", false)
scala> df2.show
+--------------------+------------+--------------------+
|                   a|           b|                   c|
+--------------------+------------+--------------------+
| 9223372036854775807| 9.223372E18|9.223372036854776E18|
|-9223372036854775808|-9.223372E18|-9.22337203685477...|
+--------------------+------------+--------------------+

scala> spark.conf.set("spark.rapids.sql.enabled", true)
scala> df2.show

+--------------------+---------------+---------------+
|                   a|              b|              c|
+--------------------+---------------+---------------+
| 9223372036854775807| 9.223372037E18| 9.223372037E18|
|-9223372036854775808|-9.223372037E18|-9.223372037E18|
+--------------------+---------------+---------------+
```

Reply:
This is an existing issue. I just checked with Spark-3.4 using spark-rapids version 23.08 

```
val df = Seq(9223372036854775807L, -9223372036854775808L).toDF("a").repartition(2)
val df2 = df.withColumn("b", expr("cast(a as float)")).withColumn("c", expr("cast(a as double)"))

spark.conf.set("spark.rapids.sql.enabled", false)
df2.show

+--------------------+------------+--------------------+
|                   a|           b|                   c|
+--------------------+------------+--------------------+
| 9223372036854775807| 9.223372E18|9.223372036854776E18|
|-9223372036854775808|-9.223372E18|-9.22337203685477...|
+--------------------+------------+--------------------+

spark.conf.set("spark.rapids.sql.enabled", true)
df2.show

+--------------------+---------------+---------------+
|                   a|              b|              c|
+--------------------+---------------+---------------+
| 9223372036854775807| 9.223372037E18| 9.223372037E18|
|-9223372036854775808|-9.223372037E18|-9.223372037E18|
+--------------------+---------------+---------------+
```

Reply:
This is also giving me a hard time to write the unit tests for float types

Reply:
This is already being tracked in https://github.com/NVIDIA/spark-rapids/issues/4204. I have added this example to the ticket.